xbmc/addons/Addon.cpp

```diff
@@ -179,76 +234,79 @@ void CAddon::SaveSettings(void)
 
   // create the XML file
   CXBMCTinyXML doc;
-  if (SettingsToXML(doc))
-    doc.SaveFile(m_userSettingsPath);
+  if (SettingsToXML(instance, doc))
+    doc.SaveFile(data.m_userSettingsPath);
 
-  m_hasUserSettings = true;
+  data.m_hasUserSettings = true;
 
   //push the settings changes to the running addon instance
-  CServiceBroker::GetAddonMgr().ReloadSettings(ID());
+  CServiceBroker::GetAddonMgr().ReloadSettings(ID(), instance);
 #ifdef HAS_PYTHON
   CServiceBroker::GetXBPython().OnSettingsChanged(ID());
 #endif
 }
 
-std::string CAddon::GetSetting(const std::string& key)
+std::string CAddon::GetSetting(uint32_t instance, const std::string& key)
```

Comment:
Unfortunately not, because the called `LoadSettings` always contains write calls.

It might be possible if the setting loading path is completely revised and comes with another call. But then it has to be checked very well so that it doesn't cause problems in any Python addon.